### PR TITLE
Fail to transmit whole resource value in notification message if it r…

### PR DESCRIPTION
**Pull request by anttiylitokola**
…equires blockwise transfer

 - Fixes error reported in https://github.com/ARMmbed/mbed-client/issues/481

### CODE REVIEW

File: source/sn_coap_protocol.c [hunk 1]
```diff
@@ -641,11 +639,21 @@ sn_coap_hdr_s *sn_coap_protocol_parse(struct coap_s *handle, sn_nsdl_addr_s *src
                 break;
             }
         }
-
+        /* Remove from the list if not an notification message.
+         * Initial notification message is needed for sending rest of the blocks (GET request).
+        */
+        bool remove_from_the_list = false;
         if (stored_blockwise_msg_temp_ptr) {
```
Comment (JaniSuonpera) [inline source/sn_coap_protocol.c]: This doable with one if
`if (stored_blockwise_msg_temp_ptr  &&
stored_blockwise_msg_temp_ptr->coap_msg_ptr && 
...
`

File: source/sn_coap_protocol.c [hunk 2]
```diff
@@ -1919,6 +1926,22 @@ static sn_coap_hdr_s *sn_coap_handle_blockwise_message(struct coap_s *handle, sn
...
+                */
+                if (src_coap_blockwise_ack_msg_ptr->options_list_ptr &&
+                    src_coap_blockwise_ack_msg_ptr->options_list_ptr->observe != COAP_OBSERVE_NONE) {
+                    if (received_coap_msg_ptr->token_len && src_coap_blockwise_ack_msg_ptr->token_ptr) {
+                        handle->sn_coap_protocol_free(src_coap_blockwise_ack_msg_ptr->token_ptr);
+                        src_coap_blockwise_ack_msg_ptr->token_ptr = NULL;
+                        src_coap_blockwise_ack_msg_ptr->token_ptr = handle->sn_coap_protocol_malloc(received_coap_msg_ptr->token_len);
+                        if (src_coap_blockwise_ack_msg_ptr->token_ptr) {
+                            memset(src_coap_blockwise_ack_msg_ptr->token_ptr, 0, received_coap_msg_ptr->token_len);
+                            memcpy(src_coap_blockwise_ack_msg_ptr->token_ptr, received_coap_msg_ptr->token_ptr, received_coap_msg_ptr->token_len);
```
Comment (TeroJaasko) [inline source/sn_coap_protocol.c]: Why set buffer to zero and then copy it over on next line?

File: source/sn_coap_protocol.c [hunk 3]
```diff
@@ -1919,6 +1926,22 @@ static sn_coap_hdr_s *sn_coap_handle_blockwise_message(struct coap_s *handle, sn
...
 
+                /* Update token to match one which is in GET request.
+                 * This is needed only in case of notification message.
+                */
+                if (src_coap_blockwise_ack_msg_ptr->options_list_ptr &&
+                    src_coap_blockwise_ack_msg_ptr->options_list_ptr->observe != COAP_OBSERVE_NONE) {
+                    if (received_coap_msg_ptr->token_len && src_coap_blockwise_ack_msg_ptr->token_ptr) {
+                        handle->sn_coap_protocol_free(src_coap_blockwise_ack_msg_ptr->token_ptr);
+                        src_coap_blockwise_ack_msg_ptr->token_ptr = NULL;
+                        src_coap_blockwise_ack_msg_ptr->token_ptr = handle->sn_coap_protocol_malloc(received_coap_msg_ptr->token_len);
```
Comment (TeroJaasko) [inline source/sn_coap_protocol.c]: This will give a coverity warning for useless assignment.

**Comment by TeroJaasko** [comment]
Please rebase this branch with master before merging to get cleaner history, preferably also squash 0120090b9d2 to 44554382299e.

**Pull request merged**